sound/soc/sof/ipc4-topology.c

```diff
@@ -232,6 +232,45 @@ static int sof_ipc4_setup_process_set_cdata(struct snd_sof_widget *swidget)
 	return 0;
 }
 
+static const struct sof_ipc4_audio_format
+*sof_ipc4_get_sink_pin_fmt(struct snd_sof_widget *swidget, int pin_id)
+{
+	struct sof_ipc4_control_data *control_data;
+	struct sof_ipc4_process *process;
+	u32 module_lib_id;
+
+	module_lib_id = sof_ipc4_get_module_lib_id(swidget->module_info);
+	/* module_lib_id == 0 if module is built-in */
+	if (module_lib_id == 0) {
+		struct sof_ipc4_base_module_cfg *base = swidget->private;
+
+		/* For built-in modules, base module config format is used for all sink pins */
+		return &base->audio_fmt;
+	}
+
+	/*
+	 * External modules are supported with process module ops,
+	 * swidget->private is process module structure here.
+	 */
+	process = swidget->private;
```

Comment:
that comment above doesnt guarantee it is in fact a process type module. Can you please add a check if (swidget->id == snd_soc_dapm_effect)  here?